cmd/relay-gossip/client/client.go

```diff
@@ -115,6 +129,52 @@ func (c *Client) Sub(ch chan drand.PublicRandResponse) UnsubFunc {
 	}
 }
 
+func (c *Client) Watch(ctx context.Context) <-chan client.Result {
+	innerCh := make(chan drand.PublicRandResponse)
+	outerCh := make(chan dclient.Result)
+	end := c.Sub(innerCh)
+
+	go func() {
+		for {
+			select {
+			case resp, ok := <-innerCh:
+				if !ok {
+					close(outerCh)
+					return
+				}
+				outerCh <- &result{resp.Round, resp.Randomness, resp.Signature}
```

Comment:
```suggestion
				select {
				case outerCh <- &result{resp.Round, resp.Randomness, resp.Signature}:
				default:
					log.Warn("randomness notification dropped due to a full channel")
				}
```

Reply:
This is a bit controversial. Since client is a library, ideally you want to leave it to the user to drop messages. (For instance, the TCP stack works that way.) I'm happy to make the change, but we should watch out for this log message.